Library/Homebrew/dependency.rb

```diff
@@ -125,7 +148,14 @@ def merge_repeats(all)
         deps = grouped.fetch(name)
         dep  = deps.first
         tags = deps.flat_map(&:tags).uniq
-        dep.class.new(name, tags, dep.env_proc)
+        taps = deps.map(&:owner_tap).flatten.uniq
+        unless taps.length == 1
```

Comment:
You may want to remove `nil` tap here. I think it's fine to have a nil tap and another tap together. Right?